### PR TITLE
CM_Params::encode() should recursively encode values returned by CM_ArrayConvertible::toArray()

### PR DESCRIPTION
**Pull request by alexispeter**
Seems that this functionality was removed in https://github.com/cargomedia/cm/pull/2187

### CODE REVIEW

**Comment by alexispeter** [comment]
please review @njam

File: library/CM/Params.php [hunk 1]
```diff
@@ -599,7 +599,7 @@ public static function encode($value, $json = null) {
             $result = ['_class' => get_class($value)];
             if ($value instanceof CM_ArrayConvertible) {
                 $array = $value->toArray();
-                $result = array_merge($result, $array);
+                $result = array_merge($result, array_map('self::encode', $array));
```
Comment (tomaszdurka) [inline library/CM/Params.php]: Maybe to not duplicate we could:
```php
$result = array_merge($result, self::encode($array));
```
?

**Comment by tomaszdurka** [comment]
@njam I took the liberty to review.
Hope it's fine with you @alexispeter :)

All/rest looks good.

File: library/CM/Params.php (hunk 1 above)
Comment (alexispeter) [inline library/CM/Params.php]: Aren't you scared of reaching the nesting limit too fast? :>

**Comment by alexispeter** [comment]
thanks for the review

**Comment by tomaszdurka** [comment]
lgtm

**Pull request merged**